src/shared/api/tmdbApi.js

```diff
@@ -18,3 +19,54 @@ export const fetchMovies = async (page = 1) => {
 		return [];
 	}
 };
+
+const fetchGenres = async () => {
+	try {
+		const response = await axios.get(`${BASE_URL}/genre/movie/list`, {
+			params: {
+				api_key: API_KEY,
+				language: 'en-US',
+			},
+		});
+		return response.data.genres;
+	} catch (error) {
+		console.error('Error fetching genres:', error);
+		throw error;
+	}
+};
+
+const fetchMoviesByGenre = async genreId => {
+	try {
+		const response = await axios.get(`${BASE_URL}/discover/movie`, {
+			params: {
+				api_key: API_KEY,
```

Comment:
instead of repeating the base url and api_key in every function, configure a default axios instance to reduce repetition and make the code cleaner